mysql/lib/client.ts

```diff
@@ -96,61 +93,41 @@ export class AwsMySQLClient extends AwsClient {
     );
   }
 
-  async updateSessionStateReadOnly(readOnly: boolean): Promise<Query | void> {
-    const result = await this.targetClient.queryWithTimeout(`SET SESSION TRANSACTION READ ${readOnly ? "ONLY" : "WRITE"}`);
-
-    this._isReadOnly = readOnly;
-    this.pluginService.getSessionStateService().setupPristineReadOnly();
-    this.pluginService.getSessionStateService().setReadOnly(readOnly);
-    return result;
-  }
-
   async setReadOnly(readOnly: boolean): Promise<Query | void> {
-    const result = await this.readOnlyQuery({ sql: `SET SESSION TRANSACTION READ ${readOnly ? "ONLY" : "WRITE"}` });
-    this._isReadOnly = readOnly;
     this.pluginService.getSessionStateService().setupPristineReadOnly();
-    this.pluginService.getSessionStateService().setReadOnly(readOnly);
+    const result = await this.queryWithoutUpdate({ sql: `SET SESSION TRANSACTION READ ${readOnly ? "ONLY" : "WRITE"}` });
+    this.pluginService.getSessionStateService().updateReadOnly(readOnly);
     return result;
   }
 
   isReadOnly(): boolean {
-    return this._isReadOnly;
+    return this.pluginService.getSessionStateService().getReadOnly();
   }
 
   async setAutoCommit(autoCommit: boolean): Promise<Query | void> {
-    if (autoCommit === this.getAutoCommit()) {
-      return;
-    }
-
     this.pluginService.getSessionStateService().setupPristineAutoCommit();
-    this.pluginService.getSessionStateService().setAutoCommit(autoCommit);
 
-    this.isAutoCommit = autoCommit;
     let setting = "1";
     if (!autoCommit) {
       setting = "0";
     }
-    return await this.query({ sql: `SET AUTOCOMMIT=${setting}` });
+    const result = await this.queryWithoutUpdate({ sql: `SET AUTOCOMMIT=${setting}` });
+    this.pluginService.getSessionStateService().setAutoCommit(autoCommit);
+    return result;
   }
 
   getAutoCommit(): boolean {
-    return this.isAutoCommit;
+    return this.pluginService.getSessionStateService().getAutoCommit();
   }
 
   async setCatalog(catalog: string): Promise<Query | void> {
-    if (catalog === this.getCatalog()) {
-      return;
-    }
-
     this.pluginService.getSessionStateService().setupPristineCatalog();
+    await this.queryWithoutUpdate({ sql: `USE ${catalog}` });
```

Comment:
Need to check for empty/blank catalog. SQL statement will be invalid in this case.